src/popover.ts

```diff
@@ -24,10 +24,49 @@ function patchAttachShadow(callback: (shadowRoot: ShadowRoot) => void) {
   };
 }
 
+function patchSelectorFn<K extends string>(
+  object: Record<PropertyKey & K, unknown>,
+  name: K,
+  mapper: (selector: string) => string,
+) {
+  const original = object[name] as (selectors: string) => NodeList;
+  Object.defineProperty(object, name, {
+    value(selector: string) {
+      return original.call(this, mapper(selector));
+    },
+  });
+}
+
+const nonEscapedOpenSelector = /(^|[^\\]):open\b/g;
+const nonEscapedClosedSelector = /(^|[^\\]):closed\b/g;
+
 export function apply() {
   const visibleElements = new WeakSet<HTMLElement>();
   let lastFocusedElement: HTMLElement | null = null;
 
+  function rewriteSelector(selector: string) {
+    return selector
```

Comment:
Adding an early return here will speed the unhappy path up a bit:

```suggestion
    if (!selector.includes(':open')) {
      return selector;
    }
    return selector
```